### PR TITLE
docs: update debug mode docs

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

### Description

Updates "Debug mode" page and moves it to the "Development" section instead of the architecture docs.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
// @hf400159 @kwanhur @avinal @yzeng25 @juzhiyuan Please review when you are available.

File: docs/en/latest/debug-mode.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+You can use APISIX's debug mode to troubleshoot your configuration.
```
Comment (yzeng25) [inline docs/en/latest/debug-mode.md:30]:
```suggestion
You can use APISIX's debug mode to troubleshoot your configurations.
```

File: docs/en/latest/debug-mode.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+You can enable the basic debug mode by adding this line to your debug configuration file (`conf/debug.yaml`):
+
+```yaml title="conf/debug.yaml"
+basic:
+  enable: true
+```
+
+:::note
+
+For APISIX releases priot to v2.10, basic debug mode is enabled by setting `apisix.enable_debug = true` in your configuration file (`conf/config.yaml`).
```
Comment (yzeng25) [inline docs/en/latest/debug-mode.md:43]: typo
```suggestion
For APISIX releases prior to v2.10, basic debug mode is enabled by setting `apisix.enable_debug = true` in your configuration file (`conf/config.yaml`).
```

File: docs/en/latest/debug-mode.md (hunk 1 above)
Comment (navendu-pottekkat) [reply docs/en/latest/debug-mode.md:30]: A deployment of APISIX will have a configuration. Considering a configuration for APISIX as a whole. Wouldn't it be better to keep it singular itself?

File: docs/en/latest/debug-mode.md (hunk 2 above)
Comment (navendu-pottekkat) [reply docs/en/latest/debug-mode.md]: Updated.

File: docs/en/latest/debug-mode.md (hunk 1 above)
Comment (yzeng25) [reply docs/en/latest/debug-mode.md:30]: Yeah you are right.

**Review by yzeng25 — DISMISSED** [review]
LGTM

File: docs/en/latest/debug-mode.md (hunk 1 above)
Comment (bisakhmondal) [inline docs/en/latest/debug-mode.md:30]:
```suggestion
You can use APISIX debug mode to troubleshoot your configuration.
```

With another heads up, something like
```suggestion
You can use APISIX debug mode to troubleshoot your configuration. Currently we support the following debug modes
```

**Review by bisakhmondal — COMMENTED** [review]
Some minor nits, LGTM

**Review by juzhiyuan — DISMISSED** [review]
Thanks!

**Review by Baoyuantop — APPROVED** [review]

**Review by guoqqqi — DISMISSED** [review]

**Comment by guoqqqi** [comment]
Hello, @bisakhmondal There are a number of conflicts that need to be resolved, Can we fix them first?

**Comment by navendu-pottekkat** [comment]
@guoqqqi Fixed the conflict.

**Review by guoqqqi — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**